include/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth.hpp

```diff
@@ -0,0 +1,136 @@
+
+                                Earth                                       (   const   Earth::Type&                aType,
+                                                                                const   Directory&                  aDataDirectory                              =   Directory::Undefined() ) ;
+
+        /// @brief              Copy constructor
+        ///
+        /// @param              [in] anEarthAtmosphericModel An Earth atmospheric model
+
+                                Earth                                       (   const   Earth&                      anEarthAtmosphericModel                     ) ;
+
+        /// @brief              Copy assignment operator
+        ///
+        /// @param              [in] anEarthAtmosphericModel An Earth atmospheric model
+        /// @return             Reference to Earth atmospheric model
+
+        Earth&                  operator =                                  (   const   Earth&                      anEarthAtmosphericModel                     ) ;
+
+        /// @brief              Destructor
+
+                                ~Earth                                      ( ) ;
+
+        /// @brief              Clone the Earth atmospheric model
+        ///
+        /// @return             Pointer to Earth atmospheric model
+
+        virtual Earth*          clone                                       ( ) const override ;
+
+        /// @brief              Get atmospheric model type
+        ///
+        /// @return             Atmospheric model type
+
+        Earth::Type             getType                                     ( ) const ;
+
+        /// @brief              Get the atmospheric density value at a given position and instant
+        ///
+        /// @param              [in] aPosition A Position
+        /// @param              [in] anInstant An Instant
+        /// @return             Atmospheric density value [kg.m^-3]
+
+        virtual Real            getDensityAt                                (   const   Position&                   aPosition,
+                                                                                const   Instant&                    anInstant                                   ) const override ;
+
+        /// @brief              Get the atmospheric density value at a given position and instant
+        ///
+        /// @param              [in] aLLA A position, expressed as latitude, longitude, altitude [deg, deg, m]
+        /// @param              [in] anInstant An instant
+        /// @return             Atmospheric density value [kg.m^-3]
+
+        Real                    getDensityAt                                (   const   LLA&                        aLLA,
+                                                                                const   Instant&                    anInstant                                   ) const ;
+
+
+
+    private:
+
+        class Impl ;
+        class ExponentialImpl ;
```

Comment:
@lucas-bremond ended up needing to keep this here. Since `Impl` is private, I think`ExponentialImpl` cannot subclass it without being part of the same class.